### DISCUSSION TITLE
HLS Stream bandwidth issue.

### DISCUSSION THREAD

**Discussion by Oliver-Mardle** [savonet/liquidsoap#4000] [User Support]
I have an mpeg-ts HLS stream setup at 48,000 & 192k. However when listening to the stream, the bandwidth is showing 211.2k...
Am I missing something in the config?

```
radio = input.pulseaudio(id="Input", device="alsa_input.pci-0000_01_00.0.multichannel-input", clock_safe=false)

aac = %ffmpeg(format="mpegts",
  %audio(
    codec="aac",
    channels=2,
    ar=48000,
    b="192k"
  ))

streams = [("aac", aac)]

def segment_name(~position,~extname,stream_name) =
  timestamp = int_of_float(gettimeofday())
  duration = 2
  "#{stream_name}_#{duration}_#{timestamp}_#{position}.#{extname}"
end

output.file.hls(playlist="live.m3u8",
  "/home/npf/s3-mount/world-service/ENASE",
  segment_duration=1.0,
  segments=20,
  segment_name=segment_name,
  streams,
  radio
  )

```

**Comment by toots** [comment]
Hi! MpegTS, by definition of the standard, is known to pad data with extra bits, I wouldn't be surprised if that could be your issue here.

You might want to test with `mp4` which is supposed to be better at that (but with extra complexity).

**Comment by Warblefly** [comment]
If it is of any use, here is my code using adts that does not inject
padding. The resampling at 32000Hz is another trick to reduce bitstream
bandwidth. I wrote this after discovering that another format was adding
the dummy data you noticed.

output.file.hls(
        playlist="live.m3u8",
        fallible=true,
        segment_name=segment_name,
        segment_duration=10.0,
        segments=20,
        segments_overhead=20,
        persist_at="/home/liquidsoap/audio/state_low.config",
        "/home/liquidsoap/audio",
                [
                (
                        "HEAAC+",
                        %ffmpeg(
                                format="adts",
                                %audio(
                                        codec="libfdk_aac",
                                        samplerate=32000,
                                        vbr=1,
                                        afterburner=1,
                                        profile="aac_he_v2"
                                        )
                        ).{bandwidth=30000,id3_version=3})],
        radio_comp
        )
